File: HappeningService/src/main/java/io/duryskuba/interestmatcher/HappeningService/resource/LocationDTO.java
package io.duryskuba.interestmatcher.HappeningService.resource;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Data
public class LocationDTO {

    private String locationId;
    private String country;
    private String city;
    private String street;

    private String locationStr;

    private double lat;
    private double lan;
}

File: LocationService/src/main/kotlin/io/duryskuba/interestmatcher/LocationService/resource/LocationDTO.kt
package io.duryskuba.interestmatcher.LocationService.resource

import java.util.*

data class LocationDTO (

        var locationId: String? = null,
        var country: String? = null,
        var city: String? = null,
        var street: String? = null,
        var lat: Double? = null,
        var lan: Double? = null,
        var locationStr: String? = null //+ valid address?
) {


    constructor(dto: LocationDTO) : this() {
        this.locationId = dto.locationId
        this.country = dto.country
        this.street = dto.street
        this.city = dto.city
        this.lat = dto.lat
        this.lan = dto.lan
        this.locationStr = dto.locationStr
    }
}